### PR TITLE
Be schedule endpoints

### PR DESCRIPTION
**Pull request by johnp418**
WIP: Working on frontend logic

Schedule endpoint added

### CODE REVIEW

File: backend/src/main/resources/data.sql [hunk 1]
```diff
@@ -5,15 +5,15 @@ VALUES
 
 INSERT INTO category(id, category_limit, create_datetime, name, parent_category_id, user_id)
 VALUES 
-(1, 200.00, now(), 'TRANSPORTATION', null, 1),
-(2, 400.00, now(), 'FOOD', null, 1),
-(3, 130.00, now(), 'ENTERTAINMENT', null, 1),
+(1, 200.00, now(), 'Transportation', null, 1),
```
Comment (mwkim0919) [inline backend/src/main/resources/data.sql]: Thank you!

File: backend/src/test/java/com/ez/ezbackend/budget/service/ScheduleServiceTest.java [hunk 2]
```diff
@@ -20,6 +28,95 @@
...
+  @Test(expected = EzIllegalRequestException.class)
+  public void test_deleteSchedulesForUser_empty_scheduleIds() {
+    Set<Long> scheduleIds = new HashSet<>();
+    scheduleService.deleteSchedulesForUser(scheduleIds, 1L);
+  }
+
+  @Test
+  @DirtiesContext
+  public void test_deleteSchedulesForUser() {
+    Set<Long> scheduleIds = new HashSet<>(Arrays.asList(1L,2L,3L));
```
Comment (mwkim0919) [inline backend/src/test/java/com/ez/ezbackend/budget/service/ScheduleServiceTest.java]: nitpicking.. `1L,2L,3L` spaces after commas

File: backend/src/main/java/com/ez/ezbackend/shared/enums/UserRole.java [hunk 3]
```diff
@@ -9,5 +9,5 @@
   ADMIN("ADMIN"),
   USER("USER");
 
-  private final String text;
+  private final String value;
```
Comment (mwkim0919) [inline backend/src/main/java/com/ez/ezbackend/shared/enums/UserRole.java]: Thank you

File: backend/src/main/java/com/ez/ezbackend/budget/response/ScheduleResponse.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  public static ScheduleResponse convertFromSchedule(Schedule schedule) {
+    Category category = schedule.getCategory();
+    return ScheduleResponse.builder()
+        .id(schedule.getId())
+        .categoryId(category != null ? category.getId() : null)
+        .categoryName(category != null ? category.getName(): null)
+        .description(schedule.getDescription())
+        .withdraw(schedule.getWithdraw())
+        .deposit(schedule.getDeposit())
+        .startDate(LocalDate.now())
```
[the remark this answers is not in the document]
Comment (johnp418) [reply backend/src/main/java/com/ez/ezbackend/budget/response/ScheduleResponse.java]: Yeah this was a typo. Fixed

File: backend/src/main/java/com/ez/ezbackend/budget/response/ScheduleResponse.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  @JsonFormat(pattern = "yyyy-MM-dd")
+  @JsonSerialize(using = LocalDateSerializer.class)
+  private LocalDate nextRecurringDate;
+
+  @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+  @JsonSerialize(using = LocalDateTimeSerializer.class)
+  private LocalDateTime createDatetime;
+
+  private String recurringPattern;
```
[the remark this answers is not in the document]
Comment (johnp418) [reply backend/src/main/java/com/ez/ezbackend/budget/response/ScheduleResponse.java]: Fixed.

File: backend/src/main/java/com/ez/ezbackend/budget/response/ScheduleResponse.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  public static ScheduleResponse convertFromSchedule(Schedule schedule) {
+    Category category = schedule.getCategory();
+    return ScheduleResponse.builder()
+        .id(schedule.getId())
+        .categoryId(category != null ? category.getId() : null)
+        .categoryName(category != null ? category.getName(): null)
```
[the remark this answers is not in the document]
Comment (johnp418) [reply backend/src/main/java/com/ez/ezbackend/budget/response/ScheduleResponse.java]: Good catch

File: backend/src/main/java/com/ez/ezbackend/budget/repository/ScheduleRepository.java [hunk 7]
```diff
@@ -1,14 +1,22 @@
...
 
 public interface ScheduleRepository extends JpaRepository<Schedule, Long> {
-  @Query("SELECT s FROM Schedule s JOIN FETCH s.user JOIN FETCH s.category WHERE s.nextRecurringDate <= :date")
+  @Query("SELECT s FROM Schedule s JOIN FETCH s.user LEFT JOIN FETCH s.category WHERE s.user = :user")
+  List<Schedule> findByUser(@Param("user") User user);
+
+  @Query("SELECT s FROM Schedule s WHERE s.id IN :ids AND s.user = :user")
+  List<Schedule> findScheduleByIdsAndUser(@Param("ids") Set<Long> ids, @Param("user") User user);
+
+  @Query("SELECT s FROM Schedule s JOIN FETCH s.user LEFT JOIN FETCH s.category WHERE s.nextRecurringDate <= :date")
```
[the remark this answers is not in the document]
Comment (johnp418) [reply backend/src/main/java/com/ez/ezbackend/budget/repository/ScheduleRepository.java]: It passes every tests without it. I'm going to take it out.

**Pull request merged**